backend/associations/serializers/election.py

```diff
@@ -114,34 +128,9 @@ def validate_against_instance(cls, instance, validated_data):
     def update(self, election, validated_data):
         self.validate_against_instance(election, validated_data)
 
-        if "association" in validated_data:
-            validated_data.pop("association")
-
-        if "registered_voters" in validated_data:
-            # Replace the registered voters if provided.
-            registered_voters = validated_data.pop("registered_voters")
-            election.registered_voters.set(registered_voters)
-
-        if "choices" in validated_data:
-            new_choices_names = set([c["name"] for c in validated_data.pop("choices")])
-
-            # Get all the old choices.
-            old_choices_names = set(
-                [c[0] for c in election.choices.values_list("name")]
-            )
-
-            # Remove the old choices not in the new choices.
-            Choice.objects.filter(
-                election=election, name__in=old_choices_names - new_choices_names
-            ).delete()
-
-            # Add the new choices not yet in the Choice table.
-            Choice.objects.bulk_create(
-                [
-                    Choice(election=election, name=choice_name)
-                    for choice_name in new_choices_names - old_choices_names
-                ]
-            )
+        for field in ("association", "voters", "choices"):
+            if field in validated_data:
+                validated_data.pop(field)
```

Comment:
Maybe they should be in `read_only_fields` then ? (at least for `voters` and `choices` since they are removed during update and create)

Reply:
They are used in create (but not in update).